### ISSUE TITLE
Allow user to quickly change node data on currently selected node with a select option

### ISSUE THREAD

**Issue by Piotr-Aueternum** [don-tnowe/godot-worldmap-builder#10]
Proposal as the next iteration of #9 which is much more convenient option to change nodes. I think that feature could be later adopted in Adding new nodes with [+] button around the circle.

Context:
Currently I'm working on Passive Skill Tree that will have 100s of nodes I found it a little difficult to change them one by one in WorldMapGraph/Path/Path Nodes while having to hover to see node index and to switch between pages

Scenario:
- As user of plugin I should be allowed to setup resources directory in WorldMapView as global config
- And then as user when node is selected I should see select input with currently selected skill resource
- And then as user when select is focused I should see list of skills in given resources directory path with scrollable content
- And then as user I should be able to select a specific skill resource from the list provided by resources directory path
- And then as user I should see updated node
![image](https://github.com/user-attachments/assets/40955c57-bb60-4ff1-969d-ffa25c63b532)

**Comment by don-tnowe** [comment]
Yeah, this is useful! It can actually be simpler: a popup list is already there if `base_type` is set, picking Resources of the type from the entire project with a search bar. The trick might be in raising awareness of this option, without adding an extra button because on my end the picker in your pull request is already quite small?

![изображение](https://github.com/user-attachments/assets/e96b9359-3886-4f4a-b06c-1f9181cb4377)

I will experiment with opening it when left-clicking the filename.

**Comment by Piotr-Aueternum** [comment]
You are right! `prop_editor.base_type = 'WorldmapNodeData'` in 'Quick Load' lists all available Resources of type.
Do you think it is possible to trigger it on [Click] of resource name instead of the option in dropdown? Right now for some reason nothing happens when you click the resource name only the dropdown works.

I double checked the `custom_min_size` I'm using `v4.3.stable.official [77dcf97d8]`

**Comment by don-tnowe** [comment]
> Do you think it is possible to trigger it on [Click] of resource name

No. And I just tried it: the Quick Load window immediately closes when the `id_pressed` signal is forcibly emitted. Another workaround would be needed, but I still managed to open the popup.

**Comment by Piotr-Aueternum** [comment]
In version 4.4 something were improved and broken a bit.
When following line is uncommented:
`prop_editor.get_child(2).id_pressed.emit.call_deferred(1)`
It will open quick load but when you try to close with X you have to do it twice.
Also noticed that selection is broken on 4.4 version for quickload in general but works for normal load so it might be irrelevant to opening the quickload on click.

I was thinking about another improvement maybe even better than selecting node. What you think about drag and drop? It would be much faster to drag and drop resource into tree to put or replace existing one. I can create separate issue for that if needed I'll try take a look into that as improvement.

**Comment by don-tnowe** [comment]
Interesting. Are there any errors in the console when trying to open or close the Quick Load popup?

On the drag-and-drop: great idea, there could be problems trying to detect if there's any resource being dragged but I might already have a solution in mind. Thanks!

**Comment by don-tnowe** [comment]
Aaaand the drag-and-drop should be done! Should be up in the main branch.

It uses a brute-force search for finding the nearest node and draws markers for all nodes visible, but this shouldn't be that much of a problem for "100s of nodes"... Could be for 1000s, though!

The "needs two clicks/key presses to close" issue is indeed bizarre. Documented in #11

**Comment by don-tnowe** [comment]
Since the "click button to open quick load" feature seems to work with 4.4's new Quick Load window (*thank you for reporting it!*), the main feature of this issue should be complete. Thanks!

**Comment by Piotr-Aueternum** [comment]
> Aaaand the drag-and-drop should be done! Should be up in the main branch.
> 
> It uses a brute-force search for finding the nearest node and draws markers for all nodes visible, but this shouldn't be that much of a problem for "100s of nodes"... Could be for 1000s, though!
> 
> The "needs two clicks/key presses to close" issue is indeed bizarre. Documented in [#11](https://github.com/don-tnowe/godot-worldmap-builder/issues/11)

Wow it works so smooth right now to change the nodes it's orders of magnitude better than I expected!
Great work, you are awesome!
